crates/node/src/lib.rs

```diff
@@ -366,11 +386,32 @@ impl Node {
             .await
             != 0
         {
+            let identities = self.ctx_manager.get_context_owned_identities(context.id)?;
+
+            let Some(our_identity) = identities.into_iter().choose(&mut thread_rng()) else {
+                bail!("no identities found for context: {}", context.id);
+            };
+
+            let our_sending_key = self
+                .ctx_manager
+                .get_own_signing_key(&context.id, &our_identity)?;
+
+            let shared_key = SharedKey::new(
+                &our_sending_key,
+                &VerifyingKey::from_bytes(&executor_public_key)?,
+            );
```

Comment:
we already have `author_id`, shared key should just be the sender key. You can add a method to derive a `SharedKey` without DHKE (since it's already pre-shared)

Reply:
this is not resolved